server/plugin/template.go

```diff
@@ -447,9 +454,15 @@ Assignees: {{range $i, $el := .Assignees -}} {{- if $i}}, {{end}}{{template "use
 It now has **{{.GetRepo.GetStargazersCount}}** stars.`))
 
 	template.Must(masterTemplate.New("newWorkflowJob").Funcs(funcMap).Parse(`
-{{template "repo" .GetRepo}} {{template "workflowJob" .GetWorkflowJob}} workflow got failed (triggered by {{template "user" .GetSender}})
-Step failed: {{.GetWorkflowJob.Steps | workflowJobFailedStep}}
+{{template "repo" .GetRepo}} {{.GetWorkflowJob.GetWorkflowName}} workflow {{if eq .GetWorkflowJob.GetConclusion "success"}}succeeded{{else}}failed{{end}} (triggered by {{template "user" .GetSender}})
+{{if eq .GetWorkflowJob.GetConclusion "failure"}}Job failed: {{template "workflowJob" .GetWorkflowJob}}
+Step failed: {{.GetWorkflowJob.Steps | workflowJobFailedStep}}{{end}}
```

Comment:
Can you explain how this works? How do we know which job failed?

Also what is going on with `{{.GetWorkflowJob.Steps | workflowJobFailedStep}}`? I'm not sure what `workflowJobFailedStep` is

Reply:
@mickmister 
The line `{{if eq .GetWorkflowJob.GetConclusion "failure"}}` checks if the workflow job’s conclusion is `"failure"`. If the job has indeed failed, the template renders additional details about the job using the workflowJob template: `Job failed: {{template "workflowJob" .GetWorkflowJob}}`.

The `workflowJobFailedStep` function processes the Steps array from the `WorkflowJob` object. It filters these steps to find and return details about the specific step that failed. This function is defined in the `funcMap` provided to the template engine, allowing it to be used as a filter within the template.